SConstruct

```diff
@@ -1,4 +1,74 @@
+from lsst.sconsUtils.utils import libraryLoaderEnvironment
+
+env = Environment(ENV=os.environ)
+env["ENV"]["OMP_NUM_THREADS"] = "1"  # Disable threading
+
+
+def getExecutableCmd(package, script, *args, directory=None):
+    """
+    Given the name of a package and a script or other executable which lies
+    within the given subdirectory (defaults to "bin"), return an appropriate
+    string which can be used to set up an appropriate environment and execute
+    the command.
+    This includes:
+    * Specifying an explict list of paths to be searched by the dynamic linker;
+    * Specifying a Python executable to be run (we assume the one on the
+      default ${PATH} is appropriate);
+    * Specifying the complete path to the script.
+    """
+    if directory is None:
+        directory = "bin"
+    cmds = [libraryLoaderEnvironment(), "python",
+            os.path.join(env.ProductDir(package), directory, script)]
+    cmds.extend(args)
+    return " ".join(cmds)
+
+
+# TODO: do we need any python here?
+# scripts.BasicSConstruct("ci_ap", disableCc=True, noCfgFile=True)
+
+num_process = GetOption('num_jobs')
+
+PKG_ROOT = env.ProductDir("ci_ap")
+DC2_ROOT = os.path.join(PKG_ROOT, "output-dc2")
+COSMOS_PDR2_ROOT = os.path.join(PKG_ROOT, "output-cosmos_pdr2")
+HITS2015_ROOT = os.path.join(PKG_ROOT, "output-hits2015")
+
+dc2_pipeline = os.path.join(env.ProductDir("ap_verify_ci_dc2"), "pipelines/ApVerifyWithFakes.yaml")
+dc2 = env.Command(DC2_ROOT, None,
+                  [getExecutableCmd("ap_verify", "ap_verify.py",
+                                    "--dataset", "ap_verify_ci_dc2",
+                                    "--pipeline", dc2_pipeline,
+                                    "--output", DC2_ROOT,
+                                    "-j", str(num_process))])
+
+cosmos_pdr2 = env.Command(COSMOS_PDR2_ROOT, None,
+                          [getExecutableCmd("ap_verify", "ap_verify.py",
+                                            "--dataset", "ap_verify_ci_cosmos_pdr2",
+                                            "--output", COSMOS_PDR2_ROOT,
+                                            "-j", str(num_process))])
+
+hits2015 = env.Command(HITS2015_ROOT, None,
+                       [getExecutableCmd("ap_verify", "ap_verify.py",
+                                         "--dataset", "ap_verify_ci_hits2015",
+                                         "--output", HITS2015_ROOT,
+                                         "-j", str(num_process))])
+
+everything = [dc2, cosmos_pdr2, hits2015]
+# Ensure that the above are not run in parallel (to avoid processor
+# contention) by claiming that they all create a (non-existent) file
+# "parallization-lock".
+env.SideEffect("parallelization-lock", everything)
+env.Alias("all", everything)
+env.Alias("dc2", dc2)
+env.Alias("cosmos", cosmos_pdr2)
+env.Alias("hits", hits2015)
+Default(everything)
+
+env.Clean(everything, [y for x in everything for y in x])
```

Comment:
This doubly-nested syntax is a little strange at first glance.  It would help to add a comment that explains why iteration over the constituents of, e.g., `dc2` is needed.

Reply:
I realize it's a bit late now, but I thought we were supposed to be modeling new CI packages on ci_imsim which uses ci_builder? ci_hsc has an open ticket for migrating it to the new CI infrastructure.

Reply:
I couldn't find any docs on how to use ci_builder (it's `doc/` directory is empty and the readme is too), and the goal here was to get something working quickly. I'd have tried to use ci_builder, if there was any information about how, but figuring out an undocumented package seemed much harder than copying a few lines from ci_hsc (which at least has a fairly well described SConstruct by now).

Reply:
And the ticket you're thinking of is https://rubinobs.atlassian.net/browse/DM-35045, which has languished since 2022. It seems like ci_builder is a bit abandoned at this point.